### DISCUSSION TITLE
`include` child documents with `params`.

### DISCUSSION THREAD

**Discussion by jospueyo** [quarto-dev/quarto-cli#8735] [Feature Requests]
### Description

I would appreciate a way to include parametrized child documents, something like this:

_child.qmd:
```
---
params:
  a: 1
---

Params is: `r params$a`
```

main.qmd:

```
---
title: "main"
format: html
---

## Include with params

{{< include _child.qmd params={a: 1}>}}

{{< include _child.qmd params={a: 2}>}}

```

I'm not sure why I used js notation to pass the params' values, it just felt right. This is a silly example, but I've found a lot of use cases for this. For instance, a quarto dashboard about water quality where each tab is the same child document just passing a different pollutant as params.

Thank you for this amazing tool!

**Comment by cscheid** [comment]
We're not going to implement something like this anytime soon. I understand why that is desirable, but it mixes together separate phases of Quarto's compilation pipeline in a way that we do not intend to.

Include shortcodes will remain textual for the foreseeable future.

**Reply by cscheid** [reply]
Your usecase can be addressed, for example, by factoring the code in _child into a library, and having that library emit the code with two.

It's not as clean as your imagined syntax, but unfortunately that feature is incompatible with Quarto's semantics: Params are a document-wide notion, and includes do not compile separately. By design, they _include_ the code into the parent document.

If you absolutely need to do something like that, you could consider `knit_child` and its configuration options, with the downside that you're restricting yourself to the knitr engine.

**Comment by mcanouil** [comment]
This has already been discussed in several other places such as:
- https://github.com/quarto-dev/quarto-cli/discussions/8271

**Reply by jospueyo** [reply]
Thank you for pointing me to that. I looked for posts containing "include" and "params", so I didn't find #8271.

**Comment by DavZim** [comment]
This would be a great addition for my use-cases.

For example, when I create a Quarto Dashboard, it is nice to separate different screens/views into different partials which I then include using the `{{< include partials/view1.qmd >}}`. Each of these partials (or the main qmd file), might want to include a partial with different parameters.
For instance, I might have a partial that inserts a plot but also a table in a tabset for my dataset, that I use for multiple plots. Being able to include partials with parameters would solve this.
